### ISSUE TITLE
TypeError: Cannot read property 'N' of undefined

### ISSUE THREAD

**Issue by daveoon** [deepstreamIO/deepstream.io#920]
```
[eval]:46126
    return this.MSG_TO_JIF[topic][messageAction][type](data)
                                                ^

TypeError: Cannot read property 'N' of undefined
    at JIFHandler.toJIF ([eval]:46126:49)
    at HTTPConnectionEndpoint._onSocketMessage ([eval]:45191:42)
    at HTTPSocketWrapper.sendMessage ([eval]:45801:12)
    at Rpc._send ([eval]:51549:14)
    at Rpc._handleResponse ([eval]:51484:10)
    at Rpc.handle ([eval]:51409:12)
    at RpcHandler.handle ([eval]:51005:21)
    at MessageDistributor.distribute ([eval]:46542:37)
    at MessageProcessor._onPermissionResponse ([eval]:46806:10)
    at ConfigPermissionHandler.canPerformAction ([eval]:47963:7)
```

I got this error and the entire DS crashed. Anyone can help?

**Comment by daveoon** [comment]
This error happened when called the RPC over HTTP by sending response.error(), change to response.send() prevent crashing the deepstream server

**Comment by yasserf** [comment]
Need to test this out still! 😅

**Comment by yasserf** [comment]
Fixed in 4.2.4

Test here: https://github.com/deepstreamIO/deepstream.io-e2e/blob/master/http/RPCs.feature#L46
